presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -190,11 +190,21 @@ public PlanNode replaceChildren(List<PlanNode> newChildren)
 
     public boolean isDecomposable(FunctionRegistry functionRegistry)
     {
-        return (getAggregations().entrySet().stream()
-                .map(entry -> functionRegistry.getAggregateFunctionImplementation(entry.getValue().getSignature()))
-                .allMatch(InternalAggregationFunction::isDecomposable)) &&
-                getAggregations().entrySet().stream()
-                        .allMatch(entry -> !entry.getValue().getCall().getOrderBy().isPresent());
+        boolean hasOrderBy = getAggregations().values().stream()
+                .map(Aggregation::getCall)
+                .map(FunctionCall::getOrderBy)
+                .anyMatch(Optional::isPresent);
```

Comment:
why not to short-circuit this and the one below?
``` 
if (hasOrderBy) { return false }
```

Reply:
It's less readable:

```java
boolean hasOrderBy = getAggregations().values().stream()
        .map(Aggregation::getCall)
        .map(FunctionCall::getOrderBy)
        .anyMatch(Optional::isPresent);

if (hasOrderBy) {
    return false;
}

boolean hasDistinct = getAggregations().values().stream()
        .map(Aggregation::getCall)
        .anyMatch(FunctionCall::isDistinct);

if (hasDistinct) {
    return false;
}

boolean decomposableFunctions = getAggregations().values().stream()
        .map(Aggregation::getSignature)
        .map(functionRegistry::getAggregateFunctionImplementation)
        .allMatch(InternalAggregationFunction::isDecomposable);

if (!decomposableFunctions) {
    return false;
}

return true;
```